internal/validators/namespace.go

```diff
@@ -107,6 +119,36 @@ func (v *Namespace) handle(req *nsRequest) admission.Response {
 	return allow("")
 }
 
+// illegalTreeLabel checks if tree labels are being created or modified
+// by any user or service account since only HNC service account is
+// allowed to do so
+func (v *Namespace) illegalTreeLabel(req *nsRequest) admission.Response {
+	msg := "Only HNC Service account can modify tree labels"
+	treeLabelID := "tree.hnc.x-k8s.io"
+
+	switch req.op {
+	case k8sadm.Update:
```

Comment:
I'd suggest combining the `Create` and `Update` tests, e.g.:

```
oldLabels := map[string]string{}
if req.oldns != nil {
  oldLabels = req.oldns.Labels
}
```

For the `Create` op, `req.oldns` will be nil so every attempt to validate that it's present will fail.

You can also do an early return for delete like:

```
if req.op == k8sadm.Delete {
  return allow("")
}
```

which saves you another level of indentation and lets you skip the switch statement.

@rjbez17 do you agree with this?

Reply:
I was following the pattern in other functions with switch statements but your suggestion would make this a whole lot simpler.

Reply:
Yeah, the pattern I'm suggesting does exist in other places but it's a bit hidden (e.g. [these three functions](https://github.com/kubernetes-sigs/hierarchical-namespaces/blob/be92981add41c07e30946e0c0da5cfa778b5eb4a/internal/validators/object.go#L215).

Reply:
Yeah I agree